src/main/java/org/opensearch/knn/index/VectorDataType.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index;
+
+import org.apache.lucene.document.FieldType;
+import org.apache.lucene.document.KnnByteVectorField;
+import org.apache.lucene.document.KnnVectorField;
+import org.apache.lucene.index.DocValuesType;
+import org.apache.lucene.index.VectorSimilarityFunction;
+import org.opensearch.index.mapper.ParametrizedFieldMapper;
+import org.opensearch.knn.index.util.KNNEngine;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import static org.opensearch.knn.common.KNNConstants.DEFAULT_VECTOR_DATA_TYPE;
+import static org.opensearch.knn.common.KNNConstants.KNN_ENGINE;
+import static org.opensearch.knn.common.KNNConstants.LUCENE_NAME;
+import static org.opensearch.knn.common.KNNConstants.VECTOR_DATA_TYPE;
+
+/**
+ * Enum contains data_type of vectors and right now only supported for lucene engine in k-NN plugin.
+ * We have two vector data_types, one is float (default) and the other one is byte.
+ */
+public enum VectorDataType {
+    BYTE("byte") {
+        /**
+         * @param dimension  Dimension of the vector
+         * @param vectorSimilarityFunction VectorSimilarityFunction for a given spaceType
+         * @return FieldType of type KnnByteVectorField
+         */
+        @Override
+        public FieldType createKnnVectorFieldType(int dimension, VectorSimilarityFunction vectorSimilarityFunction) {
+            return KnnByteVectorField.createFieldType(dimension, vectorSimilarityFunction);
+        }
+    },
+    FLOAT("float") {
+        /**
+         * @param dimension  Dimension of the vector
+         * @param vectorSimilarityFunction VectorSimilarityFunction for a given spaceType
+         * @return  FieldType of type KnnFloatVectorField
+         */
+        @Override
+        public FieldType createKnnVectorFieldType(int dimension, VectorSimilarityFunction vectorSimilarityFunction) {
+            return KnnVectorField.createFieldType(dimension, vectorSimilarityFunction);
+        }
+
+    };
+
+    private final String value;
+
+    VectorDataType(String value) {
+        this.value = value;
+    }
+
+    /**
+     * Get VectorDataType name
+     *
+     * @return  name
+     */
+    public String getValue() {
+        return value;
+    }
+
+    public abstract FieldType createKnnVectorFieldType(int dimension, VectorSimilarityFunction vectorSimilarityFunction);
+
+    /**
+     * @param knnEngine  KNNEngine
+     * @return  DocValues FieldType of type Binary and with FLOAT32 VectorEncoding
+     */
+    public FieldType buildDocValuesFieldType(KNNEngine knnEngine) {
```

Comment:
Just in case for any other vector data type that we add in future requires custom changes then we need to override this method and implement it here. So, moved this method into this class.